### DISCUSSION TITLE
Autotiling

### DISCUSSION THREAD

**Discussion by mrgreen3** [i3/i3#5490] [Q&A] [answered]
Are there any plans to add autotiling to i3? I know you can add this function in a number of ways, but feel it would be more useful than having gaps. 

Wondering if the function I need is just a keybind away ;-)

**Comment by orestisfl** [comment, marked as answer]
No, that's something that has been discussed before:
- https://github.com/i3/i3/issues/674
- https://github.com/i3/i3/issues/2106
- https://github.com/i3/i3/issues/4253

Autotiling can be achieved with scripts and many exist. Example from a quick search: https://github.com/nwg-piotr/autotiling

**Reply by mrgreen3** [reply]
Added in autotiling, have used it in the past. It works but would be nice if that function was already part of i3.  #2106 seems a reasonable request. 

Am sorry for the noise....

**Reply by medwatt** [reply]
I have some sort of autotiling going on. Check [here](https://www.youtube.com/watch?v=-2eguz6_hF4&t=1380s) to see how mine works.

**Reply by mrgreen3** [reply]
That is very interesting, your keybinds are on a whole new level. Wonder if it could be written in shell rather than Python but either way works very well and has improved your workflow immensely. One thing I would like is a way to make tiling more dynamic like DWM having a slightly larger focused window. 

Thanks so much for sharing ;-)

**Reply by medwatt** [reply]
I think writing anything other than simple scripts on bash is very tedious. I'm not familiar with dwm. What does "having a slightly larger focused window" mean? You mean every time you change focus to a window, it gets larger?

**Reply by mrgreen3** [reply]
Main window takes up 60% of screen before opening other windows to the right, have seen your other script which is amazing that does this. Am not really into floating windows on a tiler but when I do screenshots for ArchBang it would be nice to show off the effect. Again will look at your code as I am guessing having one workspace floating would be possible. 

Did try to get ChatGPT to help me with bash version was not able to communicate with i3ipc, at that point I gave in and started running https://github.com/olemartinorg/i3-alternating-layout....

Never knew i3ipc was that powerful.

**Reply by medwatt** [reply]
Spiral layout looks cool, but I don't find spiral layout useful.

**Comment by mrgreen3** [comment]
That is something I would never need or use.
